server/athenian/api/auth.py

```diff
@@ -347,32 +350,44 @@ async def _get_user_info_cached(self, token: str) -> User:
                 detail=user.get("description", str(user))))
         return User.from_auth0(**user)
 
-    async def _set_user(self, request: AthenianWebRequest, token: str) -> None:
-        token_info = await self._extract_token(token)
-        request.uid = token_info["sub"]
-        request.native_uid = request.uid.rsplit("|", 1)[1]
+    async def _set_user(self, request: AthenianWebRequest, token: str, method: str) -> None:
+        if method == "bearer":
+            token_info = await self._extract_bearer_token(token)
+            request.uid = token_info["sub"]
+        elif method == "apikey":
+            request.uid = await self._extract_api_key(token, request)
+        else:
+            raise AssertionError("Unsupported auth method: %s" % method)
+
         god = await request.sdb.fetch_one(
             select([God.mapped_id]).where(God.user_id == request.uid))
         if god is not None:
             request.god_id = request.uid
-            mapped_id = god[God.mapped_id.key]
+            if "X-Identity" in request.headers:
+                mapped_id = request.headers["X-Identity"]
+            else:
+                mapped_id = god[God.mapped_id.key]
             if mapped_id is not None:
                 request.uid = mapped_id
-                request.native_uid = mapped_id.rsplit("|", 1)[1]
                 self.log.info("God mode: %s became %s", request.god_id, mapped_id)
+
+        request.native_uid = request.uid.rsplit("|", 1)[1]
         request.is_default_user = request.uid == self._default_user_id
         sentry_sdk.add_breadcrumb(category="user", message=request.uid, level="info")
 
         async def get_user_info():
-            if god is not None and request.god_id is not None:
+            if method != "bearer" or (god is not None and request.god_id is not None):
                 return await self.get_user(request.uid)
             return await self._get_user_info(token)
 
         request.user = get_user_info
 
-    async def _extract_token(self, token: str) -> Dict[str, Any]:
+    async def _extract_bearer_token(self, token: str) -> Dict[str, Any]:
         if token == "null":
             return {"sub": self.force_user or self._default_user_id}
+        # People who understand what's going on here:
```

Comment:
😂 

Anyway reviewing the code helps a lot! But I'd definitely feel uncomfortable being listed here 😂

Reply:
haha